ipa-core/src/protocol/ipa_prf/aggregation/bucket.rs

```diff
@@ -145,34 +166,36 @@ where
 pub mod tests {
     use rand::thread_rng;
 
+    use super::move_single_value_to_bucket;
     use crate::{
-        ff::{Fp32BitPrime, Gf8Bit, Gf9Bit, U128Conversions},
+        ff::{boolean::Boolean, boolean_array::BA8, Gf8Bit, Gf9Bit, U128Conversions},
         protocol::{
             context::{Context, UpgradableContext, Validator},
-            ipa_prf::prf_sharding::bucket::move_single_value_to_bucket,
             RecordId,
         },
         rand::Rng,
-        secret_sharing::SharedValue,
+        secret_sharing::{BitDecomposed, SharedValue},
         test_executor::run,
-        test_fixture::{get_bits, Reconstruct, Runner, TestWorld},
+        test_fixture::{Reconstruct, Runner, TestWorld},
     };
 
     const MAX_BREAKDOWN_COUNT: usize = 256;
     const VALUE: u32 = 10;
 
-    async fn move_to_bucket(count: usize, breakdown_key: usize, robust: bool) -> Vec<Fp32BitPrime> {
-        let breakdown_key_bits =
-            get_bits::<Fp32BitPrime>(breakdown_key.try_into().unwrap(), Gf8Bit::BITS);
-        let value = Fp32BitPrime::truncate_from(VALUE);
+    async fn move_to_bucket(count: usize, breakdown_key: usize, robust: bool) -> Vec<BA8> {
+        let breakdown_key_bits = BitDecomposed::decompose(Gf8Bit::BITS, |i| {
```

Comment:
nit: it may be worth explaining why 8 bit value is enough here. (I believe it is related to `MAX_BREAKDOWN_COUNT` and if that's true, maybe a static assertion is possible too)